modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/downsampling/ReplaceSourceWithDownsampleIndexTask.java

```diff
@@ -0,0 +1,205 @@
+        } else {
+            // the source index exists
+            DataStream sourceParentDataStream = sourceIndexAbstraction.getParentDataStream();
+            if (sourceParentDataStream != null) {
+                assert sourceParentDataStream.getName().equals(dataStreamName)
+                    : "the backing index must be part of the provided data "
+                        + "stream ["
+                        + dataStreamName
+                        + "] but it is instead part of data stream ["
+                        + sourceParentDataStream.getName()
+                        + "]";
+                if (sourceParentDataStream.getWriteIndex().getName().equals(sourceBackingIndex)) {
+                    String errorMessage = String.format(
+                        Locale.ROOT,
+                        "index [%s] is the write index for data stream [%s] and cannot be replaced",
+                        sourceBackingIndex,
+                        sourceParentDataStream.getName()
+                    );
+                    throw new IllegalStateException(errorMessage);
+                }
+                if (sourceIndexMeta != null) {
+                    // both indices exist, let's copy the origination date from the source index to the downsample index
+                    Metadata.Builder newMetaData = Metadata.builder(state.getMetadata());
+                    IndexMetadata.Builder downsampleIndexBuilder = IndexMetadata.builder(downsampleIndexMeta);
+                    Map<String, String> lifecycleCustomMetadata = downsampleIndexMeta.getCustomData(LIFECYCLE_CUSTOM_INDEX_METADATA_KEY);
+                    Map<String, String> newCustomMetadata = new HashMap<>();
+                    if (lifecycleCustomMetadata != null) {
+                        newCustomMetadata.putAll(lifecycleCustomMetadata);
+                    }
+                    newCustomMetadata.put(REPLACEMENT_SOURCE_INDEX, sourceIndexMeta.getIndex().getName());
+                    downsampleIndexBuilder.putCustom(LIFECYCLE_CUSTOM_INDEX_METADATA_KEY, newCustomMetadata);
+
+                    if (IndexSettings.LIFECYCLE_ORIGINATION_DATE_SETTING.exists(downsampleIndexMeta.getSettings()) == false) {
+                        TimeValue sourceGenerationTime = sourceParentDataStream.getGenerationLifecycleDate(sourceIndexMeta);
+                        assert sourceGenerationTime != null : "the source index to replace must not be the write index of the data stream";
+
+                        downsampleIndexBuilder.settings(
+                            Settings.builder()
+                                .put(downsampleIndexMeta.getSettings())
+                                .put(IndexSettings.LIFECYCLE_ORIGINATION_DATE, sourceGenerationTime.millis())
+                                .build()
+                        ).settingsVersion(downsampleIndexMeta.getSettingsVersion() + 1L);
+                    }
+
+                    newMetaData.put(downsampleIndexBuilder);
```

Comment:
If we keep this, can we factor this out into a method named something like `makeClonedLifecycleState(source, dest)`?